src/_guides/language/effective-dart/usage.md

```diff
@@ -1156,6 +1233,54 @@ Future<bool> fileContainsBear(String path) async {
 }
 {% endprettify %}
 
+
+### DO test for a future when disambiguating a `FutureOr<T>` whose value type could be `Object`.
+
+Before you can do anything useful with a `FutureOr<T>`, you typically need to do
+an `is` check to see if you have a `Future<T>` or a bare `T`. If the type
+argument is some specific type like `FutureOr<int>`, it doesn't matter which
+test you use, `is int` or `is Future<int>`. Either works because those two types
+are disjoint.
+
+However, if the value type is `Object` or a type parameter that could possibly
+be instantiated with `Object` then the two types overlap. `Future<Object>`
+itself implements `Object`, so `is Object` or `is T` where `T` is some type
+parameter that could be instantiated with object returns true even when the
+object is a future. Instead, explicitly test for the `Future` case:
```

Comment:
I'm not crazy about how wordy this paragraph is either. (I'm on the fence about whether it's worth even having this guideline), but I think your suggestion is too brief. The challenging case — which is really why this guideline is useful, because the case is non-obvious — is:

```
foo<T>(FutureOr<T> f) {
  if (f is T) {
    // Assume f is not a future...
  } else {
    // Assume f is a future...
  }
}
```

This code looks correct, doesn't mention `Object`, and does work correctly for many type arguments. But if you happen to call it with `Object` as the generic method type argument, it falls over.

The prose and example try to show this.